### PR TITLE
DOC Mapping updates are deprecated for ingestion privileges

### PR DESCRIPTION
**Pull request by albertzaharovits**
I've forgot to adjust the docs and the migration notice following https://github.com/elastic/elasticsearch/pull/58784 .

The text describes the deprecated behaviour in the next subsequent minor releases, which then turns into breaking behaviour in version 8, but the code (and the accompanying  notice) that it refers to, is still due. I will raise that PR promptly.

### CODE REVIEW

File: x-pack/docs/en/security/authorization/privileges.asciidoc [hunk 1]
```diff
@@ -162,8 +162,12 @@ bulk request on an index or data stream that contains new fields that may
 be mapped rather than an explicit <<indices-put-mapping,put mapping>> request.
 
 `create`::
-Privilege to index documents. Also grants access to the update mapping
-action.
+Privilege to index documents, allowing overwriting any existing document, but
+not permitting updating one.
+_Deprecated:_ Also grants the permission to update the index mapping (but not
```
Comment (jrodewig) [inline x-pack/docs/en/security/authorization/privileges.asciidoc:167]: Just a minor formatting change.
```suggestion
not permitting updating one.
+
deprecated:[7.9] Also grants the permission to update the index mapping (but not
```

File: x-pack/docs/en/security/authorization/privileges.asciidoc [hunk 2]
```diff
@@ -174,8 +178,12 @@ privilege for an alternative.
 --
 
 `create_doc`::
-Privilege to index documents. Also grants access to the update mapping action.
-However, it does not enable a user to update existing documents.
+Privilege to index new documents, without allowing overwriting or updating
+existing ones.
+_Deprecated:_ Also grants the permission to update the index mapping (but not
```
Comment (jrodewig) [inline x-pack/docs/en/security/authorization/privileges.asciidoc:183]:
```suggestion
existing ones.
+
deprecated:[7.9] Also grants the permission to update the index mapping (but not
```

File: x-pack/docs/en/security/authorization/privileges.asciidoc [hunk 3]
```diff
@@ -205,8 +213,11 @@ Privilege to delete documents.
 Privilege to delete an index.
 
 `index`::
-Privilege to index and update documents. Also grants access to the update
-mapping action.
+Privilege to index (overwriting included) and update documents.
+_Deprecated:_ Also grants the permission to update the index mapping (but not
```
Comment (jrodewig) [inline x-pack/docs/en/security/authorization/privileges.asciidoc:217]:
```suggestion
Privilege to index (overwriting included) and update documents.
+
deprecated:[7.9] Also grants the permission to update the index mapping (but not
```

File: x-pack/docs/en/security/authorization/privileges.asciidoc [hunk 4]
```diff
@@ -252,8 +263,10 @@ privilege is primarily available for use by {kib} users.
 `write`::
 Privilege to perform all write operations to documents, which includes the
 permission to index, update, and delete documents as well as performing bulk
-operations. Also grants access to the update mapping action.
-
+operations, and allows the {ref}/dynamic-mapping.html[dynamic mapping updates]
+as a result of these.
+_Deprecated:_ Until the next major release, this also grants access to the
```
Comment (jrodewig) [inline x-pack/docs/en/security/authorization/privileges.asciidoc:268]:
```suggestion
as a result of these.
+
deprecated:[7.9] Until the next major release, this also grants access to the
```

**Review by jrodewig — APPROVED** [review]
LGTM.

I left some suggestions for the breaking changes I think need to be incorporated to fix the rendering.

The remaining suggestions are minor formatting fixes. However, the changes would be fine without them.

File: x-pack/docs/en/security/authorization/privileges.asciidoc (hunk 1 above)
Comment (albertzaharovits) [reply x-pack/docs/en/security/authorization/privileges.asciidoc:167]: Aargh, I knew about the `deprecated:[x.x]` marker.

**Comment by albertzaharovits** [comment]
Thank you @jrodewig ! I like your revised phrasing much more, very good!

File: docs/reference/migration/migrate_7_9.asciidoc [hunk 5]
```diff
@@ -70,6 +70,40 @@ context.  For example, for the `processor_conditional` context, use
...
+In {es} 8.0.0, the following privileges will no longer allow users to
+explicitly update the mapping of an index:
+
+* `create_doc`
+* `create`
+* `index`
+* `write`
+
+Additionally, in {es} 8.0.0, the following privileges will no longer allow users to
+{ref}/dynamic-mapping.html[dynamic update the mapping] of an index
```
Comment (tvernum) [inline docs/reference/migration/migrate_7_9.asciidoc:88]: maybe
```suggestion
{ref}/dynamic-mapping.html[dynamically update the mapping] of an index 
```
or
```suggestion
perform a {ref}/dynamic-mapping.html[dynamic update of the mapping] of an index 
```
?

**Review by tvernum — APPROVED** [review]
LGTM, with 1 minor suggestion

File: docs/reference/migration/migrate_7_9.asciidoc (hunk 5 above)
Comment (albertzaharovits) [reply docs/reference/migration/migrate_7_9.asciidoc]: Thank you Tim!

**Pull request merged**